### PR TITLE
[RW-2719][risk=no] Convert root CB component to React

### PR DESCRIPTION
**Pull request by dolbeew**
Converts the `CohortSearch` component from Angular to React. Should not include any design or functionality changes.

@NehaBroad to review React conversion
@freemabd to deploy UI locally to check functionality

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] This PR includes appropriate unit tests
- [x] I have run and tested this change locally

### CODE REVIEW

File: ui/src/app/cohort-search/cohort-search/cohort-search.component.tsx [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+      setTimeout(() => this.setState({updateCount: this.state.updateCount + 1}));
+    }
+
+    render() {
+      const {cohort, cohortChanged, criteria, loading, modalOpen, overview, searchContext, updateCount, updateGroupListsCount} = this.state;
+      return <React.Fragment>
+        <div ref={el => this.searchWrapper = el} style={{padding: '1rem 1rem 2rem'}}>
+        <div style={{display: 'flex', flexWrap: 'wrap', margin: '0 -0.5rem'}}>
+          <div style={colStyle('66.66667')}>
+            <div style={{display: 'flex', flexWrap: 'wrap', margin: '0 -0.5rem'}}>
```
Comment (NehaBroad) [inline ui/src/app/cohort-search/cohort-search/cohort-search.component.tsx:141]: We can use FlexRowWrap here

Comment (dolbeew) [reply ui/src/app/cohort-search/cohort-search/cohort-search.component.tsx]: Nice, didn't know these were here.

File: ui/src/app/cohort-search/cohort-search/cohort-search.component.tsx [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    componentWillUnmount() {
+      this.subscription.unsubscribe();
+      idsInUse.next(new Set());
+      currentCohortStore.next(undefined);
+      searchRequestStore.next({includes: [], excludes: [], dataFilters: []} as SearchRequest);
+    }
+
+    async showWarningModal() {
+      this.setState({modalOpen: true});
+      return await new Promise<boolean>((resolve => this.resolve = resolve));
```
Comment (NehaBroad) [inline ui/src/app/cohort-search/cohort-search/cohort-search.component.tsx:118]: Question: Whats _await new Promise<boolean>((resolve => this.resolve = resolve));_ used for

File: ui/src/app/cohort-search/cohort-search/cohort-search.component.tsx [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+          this.setState({cohort: {criteria: `{'includes':[],'excludes':[],'dataFilters':[]}`, name: '', type: ''}});
+        }
+      });
+
+      this.subscription.add(searchRequestStore.subscribe(sr => {
+        const cohortChanged = !!this.state.cohort && this.state.cohort.criteria !== JSON.stringify(mapRequest(sr));
+        this.props.setCohortChanged(cohortChanged);
+        this.setState({
+          criteria: sr,
+          overview: sr.includes.length > 0 || sr.excludes.length > 0,
```
Comment (NehaBroad) [inline ui/src/app/cohort-search/cohort-search/cohort-search.component.tsx:100]: can sr ever be null? also can we use more meaningful variable searchRequest or something?

File: ui/src/app/cohort-search/cohort-search/cohort-search.component.tsx [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+    componentDidMount() {
+      const {workspace: {id, namespace}} = this.props;
+      this.subscription = queryParamsStore.subscribe(params => {
+        /* If a cohort id is given in the route, we initialize state with
+         * it */
+        const cohortId = params.cohortId;
+        if (cohortId) {
+          this.setState({loading: true});
+          cohortsApi().getCohort(namespace, id, cohortId).then(cohort => {
+            this.setState({cohort, loading: false});
```
Comment (NehaBroad) [inline ui/src/app/cohort-search/cohort-search/cohort-search.component.tsx:84]: What happens if this api throws an error

File: ui/src/app/cohort-search/cohort-search/cohort-search.component.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    currentWorkspaceStore.next(workspaceDataStub);
+    queryParamsStore.next({
+      cohortId: 'test-id'
+    });
+  });
+  it('should render', () => {
+    const wrapper = shallow(<CohortSearch setCohortChanged={() => {}} setShowWarningModal={() => {}} setUpdatingCohort={() => {}}/>);
+    expect(wrapper).toBeTruthy();
+  });
+});
```
Comment (NehaBroad) [inline ui/src/app/cohort-search/cohort-search/cohort-search.component.spec.tsx:19]: Can we add more test like for search request with include/exclude?

Comment (dolbeew) [reply ui/src/app/cohort-search/cohort-search/cohort-search.component.spec.tsx:43]: Probably could use more still, but I added one to check that the number of groups matches the length of the includes/excludes array.

**Pull request merged**